src/plugins/discourse/fetch.js

```diff
@@ -303,6 +363,50 @@ function failForNotOk(response: Response) {
   }
 }
 
+/**
+ * Parses a "latest" topic.
+ *
+ * A "latest" topic, is a topic as returned by the /latest.json API call,
+ * and has two distinct assumptions:
+ * - bumped_at is always present.
+ * - the posters field uses IDs instead of usernames.
+ *
+ * usernamesById map used to resolve these IDs to usernames.
+ */
+function parseLatestTopic(
+  usernamesById: Map<number, string>,
+  json: any
+): Topic {
+  // TODO: this is a code smell.
+  // Currently we're using a quirk of the API code, that the first poster in the summary
+  // is always the original poster. The only other way to tell is by parsing it from the
+  // translated human description.
+  // See: https://github.com/discourse/discourse/blob/23367e79ea735598766ec6f507f6132e0bad3dba/lib/topic_query.rb#L443
+  const opUsername = usernamesById.get(json.posters[0].user_id);
+  if (opUsername == null) {
+    throw new Error(
+      `Unexpected missing OP user for ${stringify(
+        json.posters[0]
+      )} in map ${stringify(Array.from(usernamesById.entries()))}`
+    );
+  }
+
+  if (json.bumped_at == null) {
+    throw new Error(
```

Comment:
Similar to the other comment. If this works for a small forum but not a large one, it would be a Discourse bug. As in theory, this `bumped_at` field is not nullable on the API.

The reason I've included it here, is because it's actually our _only_ reliable source of `bumpedMs` I found in the current API. Should the API ever change this, it will fundamentally break the new syncing mechanism. Which is why I've made it an explicit crash, but should never happen, barring and API change or bug.